sickbeard/clients/generic.py

```diff
@@ -1,241 +1,275 @@
+        """
+        This should be overridden and should return the auth_id needed for the client
+        """
+        return None
+
+    def sinlge_episode_enable(self):
+        """
+        This should be overridden and should return True if single episode download form multiepisode torrent is enabled
+        """
+
+        return False
+
+    def manage_single_episode(self, result, is_not_downloading):
+
+        return True
+
+    def _add_torrent_uri(self, result):
+        """
+        This should be overridden should return the True/False from the client
+        when a torrent is added via url (magnet or .torrent link)
+        """
+        return False
+
+    def _get_file_list_in_torrent (self, result):
+        """
+        This should be overridden should return the True/False from the client
+        when a torrent is added via url (magnet or .torrent link)
+        """
+        return []
+
+    def _add_torrent_file(self, result):
+        """
+        This should be overridden should return the True/False from the client
+        when a torrent is added via result.content (only .torrent file)
+        """
+        return False
+
+    def _set_torrent_label(self, result):
+        """
+        This should be overridden should return the True/False from the client
+        when a torrent is set with label
+        """
+        return True
+
+    def _set_torrent_ratio(self, result):
+        """
+        This should be overridden should return the True/False from the client
+        when a torrent is set with ratio
+        """
+        return True
+
+    def _set_torrent_seed_time(self, result):
+        """
+        This should be overridden should return the True/False from the client
+        when a torrent is set with a seed time
+        """
+        return True
+
+    def _set_torrent_priority(self, result):
+        """
+        This should be overriden should return the True/False from the client
+        when a torrent is set with result.priority (-1 = low, 0 = normal, 1 = high)
+        """
+        return True
+
+    def _set_torrent_path(self, torrent_path):
+        """
+        This should be overridden should return the True/False from the client
+        when a torrent is set with path
+        """
+        return True
+
+    def _set_torrent_pause(self, result):
+        """
+        This should be overridden should return the True/False from the client
+        when a torrent is set with pause
+        """
+        return True
+
+    def _get_torrent_hash(self, result):
+
+        if result.url.startswith('magnet'):
+            result.hash = re.findall('urn:btih:([\w]{32,40})', result.url)[0]
+            if len(result.hash) == 32:
+                result.hash = b16encode(b32decode(result.hash)).lower()
+        else:
+            if not result.content:
+                logger.log('Torrent without content, try to geti it', logger.ERROR)
+                result.content = result.provider.getURL(result.url)
```

Comment:
@abeloin i've added this peace of code because it happens that result.content is empty not because the torrent is corrupt. let me know if it's fine